### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by joshcombes**
This is just a suggestion. If you don't like parts please directly edit the file.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Contributor Guidelines
+
+We value and encourage contribution from the community. To reduce
+friction in this process, we have collected some best-practices for
+contributors:
+
+* API style. We have chosen to split the API of the modules in our 
+  benchmarking package in the following way:
+  `exp_obj = generate_X_experiment( parameters)`
+  `results_obj = acquire_data_X(exp_obj)`
```
Comment (kylegulshen) [inline CONTRIBUTING.md]: Sorry I missed this in the qubit specs refactor: the majority of existing modules use acquire_X_data, similar to acquire_X_experiment. Shall we change it here and make a note for the specs module?

Comment (joshcombes) [reply CONTRIBUTING.md]: Good point. I'll change the specs module.

Comment (joshcombes) [reply CONTRIBUTING.md]: this is now in a branch so I will resolve this commet

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+* Testing. Before making a pull-request (PR), please make sure that
+  you have tested your new QCVV code (and any other modules you have 
+  changed) for mathematical correctness by including important test 
+  cases in `/tests/` and then running the tests. We use the pytest 
+  framework.
+  
+* Documentation. Public functions *must* have docstrings. Documentation 
+  should also be provided as comments to your code. Finally please include 
+  references to to the relavent papers and equations in those papers.
+  The recomended referernce style guide is:
```
Comment (notmgsk) [inline CONTRIBUTING.md]: spelling

Comment (joshcombes) [reply CONTRIBUTING.md]: :(

**Pull request merged**